normandy/recipes/exports.py

```diff
@@ -0,0 +1,120 @@
+        # is basically a tiny abstraction on top of the requests library.
+        self.client = (
+            kinto_http.Client(
+                server_url=str(settings.REMOTE_SETTINGS_URL),
+                auth=(
+                    str(settings.REMOTE_SETTINGS_USERNAME),
+                    str(settings.REMOTE_SETTINGS_PASSWORD),
+                ),
+                bucket=str(settings.REMOTE_SETTINGS_BUCKET_ID),
+                collection=self.collection_id,
+                retry=int(settings.REMOTE_SETTINGS_RETRY_REQUESTS),
+            )
+            if settings.REMOTE_SETTINGS_URL
+            else None
+        )
+
+    def check_config(self):
+        """
+        Verify that integration with Remote Settings is configured properly.
+        """
+        if self.client is None:
+            return  # no check if disabled.
+
+        required_keys = ["COLLECTION_ID", "USERNAME", "PASSWORD"]
+        for key in required_keys:
+            if not getattr(settings, f"REMOTE_SETTINGS_{key}"):
+                msg = f"set settings.REMOTE_SETTINGS_{key} to use Remote Settings integration"
+                raise ImproperlyConfigured(msg)
+
+        # Test authentication.
+        server_info = self.client.server_info()
+        is_authenticated = (
+            "user" in server_info
+            and settings.REMOTE_SETTINGS_USERNAME in server_info["user"]["id"]
+        )
+        if not is_authenticated:
+            raise ImproperlyConfigured("Invalid Remote Settings credentials")
+
+        # Test that collection is writable.
+        metadata = self.client.get_collection()
+        if server_info["user"]["id"] not in metadata["permissions"].get("write", []):
+            raise ImproperlyConfigured(
+                f"Remote Settings collection {self.collection_id} is not writable."
+            )
+
+    def publish(self, recipe):
+        """
+        Publish the specified `recipe` on the remote server by upserting a record.
+        """
+        if self.client is None:
+            return  # no-op if disabled.
+
+        record = recipe_as_record(recipe)
+        self.client.update_record(data=record)
+        self.client.patch_collection(id=self.collection_id, data={"status": "to-sign"})
+        logger.info(f"Published record '{recipe.id}' for recipe {recipe}")
+
+    def unpublish(self, recipe):
+        """
+        Unpublish the specified `recipe` by deleted its associated records on the remote server.
+        """
+        if self.client is None:
+            return  # no-op if disabled.
+
+        try:
+            self.client.delete_record(id=str(recipe.id))
+
+        except kinto_http.KintoException as e:
+            if e.response.status_code == 404:
+                logger.warning(f"The recipe '{recipe.id}' was never published. Skip.")
+                return
+            raise
+
+        self.client.patch_collection(id=self.collection_id, data={"status": "to-sign"})
```

Comment:
What affect does this line have? I assume it is related to the special signing bypass you mentioned above. Can you add a comment explaining it?